### ISSUE TITLE
Ensure cache keys using PID are unique

### ISSUE THREAD

**Issue by maxbrunet** [parca-dev/parca-agent#984]
> One small comment about something we need to revisit in the Agent as a whole is that caching PIDs can be problematic due to them being reused.

We should be able to add the `starttime` field of `/proc/[pid]/stat` to the cache key: `awk '{print $23}' "/proc/${pid}/stat"`

https://pkg.go.dev/github.com/prometheus/procfs#ProcStat.StartTime

_Originally posted by @maxbrunet in https://github.com/parca-dev/parca-agent/issues/960#issuecomment-1300932155_

There are at least two caches to look into, in the discovery manager and in the label manager

cc @javierhonduco @kakkoyun

**Comment by javierhonduco** [comment]
Yes! Exactly the approach I had in mind :). It's how I implemented a [similar logic](https://github.com/javierhonduco/rbperf/commit/272c8f3087b9f85c092ffccfbd131088e6b7d21d) in rbperf to ensure that the right process is profiled, and was planning on reworking all this in the Agent at some point
